ffcx/codegeneration/access.py

```diff
@@ -350,3 +362,48 @@ def facet_edge_vectors(self, e, mt, tabledata, num_points):
     def _pass(self, *args, **kwargs):
         """Return one."""
         return 1
+
+    def table_access(self, tabledata: UniqueTableReferenceT, entitytype: str, restriction: str,
+                     quadrature_index: L.MultiIndex, dof_index: L.MultiIndex):
+        """
+        Access element table for given entity, quadrature point, and dof index.
+
+        Args:
+            tabledata: Table data object
+            entitytype: Entity type ("cell", "facet", "vertex")
+            restriction: Restriction ("+", "-")
+            quadrature_index: Quadrature index
+            dof_index: Dof index
+        """
+        entity = self.symbols.entity(entitytype, restriction)
+        iq_global_index = quadrature_index.global_index
+        ic_global_index = dof_index.global_index
+        qp = 0  # quadrature permutation
+
+        symbols = []
+        if tabledata.is_uniform:
+            entity = L.LiteralInt(0)
+
+        if tabledata.is_piecewise:
+            iq_global_index = L.LiteralInt(0)
+
+        # FIXME: Hopefully tabledata is not permuted when applying sum
+        # factorization
+        if tabledata.is_permuted:
+            qp = self.symbols.quadrature_permutation[0]
+            if restriction == "-":
+                qp = self.symbols.quadrature_permutation[1]
```

Comment:
We should really either:
1. Use enums for restrictions in FFCx
2. Use enums for restrictions in UFL (which could have a string representation: https://docs.python.org/3/library/enum.html#enum.StrEnum).
@mscroggs what do you think?

Reply:
Should this be discussed in the UFL issue tracker?

Reply:
Yes, do you mind adding an issue?